### PR TITLE
Add pre-commit and apply fixes

### PR DESCRIPTION
**Pull request by lukas-b**
Added [http://pre-commit.com/](pre-commit) ([GitHub](https://github.com/pre-commit/pre-commit)) with a basic set of [pre-commit-hooks](https://github.com/pre-commit/pre-commit-hooks) and applied fixes.

### CODE REVIEW

**Comment by jtwang** [comment]
Hey, uh, @lukas-b, I think the general consensus of the group is that we don't want to use pre-commit in this repo. Mind popping into the gitter channel and asking?

**Comment by akx** [comment]
Having pre-commit as an opt-in is fine by me, but @sils1297 's http://coala-analyzer.org/ (via gitmate) does some of that stuff in GitHub CI...

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ For a PR to be merged, the following statements must hold true:
   not the author of the PR. Commits shall comply to the "Good Commits" standards
   outlined below.
 
-To begin contributing have a look at the open [easy issues](https://github.com/python-babel/babel/issues?q=is%3Aopen+is%3Aissue+label%3Adifficulty%2Flow) 
+To begin contributing have a look at the open [easy issues](https://github.com/python-babel/babel/issues?q=is%3Aopen+is%3Aissue+label%3Adifficulty%2Flow)
```
Comment (jtwang) [inline CONTRIBUTING.md]: Although I approve of removing trailing whitespace from this line, I think this file should be excluded from the pre-commit checks.

**Comment by jtwang** [comment]
Change looks fine to me, one minor nitpick. I'm a bit baffled by why test coverage went up though. yolo. :)

**Comment by akx** [comment]
> I'm a bit baffled by why test coverage went up though

The coverage diff says

```patch
-del messages[:]; del translations[:]; del context[:]; del locations[:];
-del flags[:]; del auto_comments[:]; del user_comments[:];
+del messages[:]
+del translations[:]
+del context[:]
+del locations[:]
+del flags[:]
+del auto_comments[:]
+del user_comments[:]
```

so that's +5 _lines_ that exist and are executed just the same :)

File: CONTRIBUTING.md (hunk 1 above)
Comment (lukas-b) [inline CONTRIBUTING.md]: Makes sense :) Will keep the change and exclude the file, thanks!

**Comment by lukas-b** [comment]
Thanks for the CR! :) 🐏🚀
For future explorers: [gitter.im/python-babel/babel](https://gitter.im/python-babel/babel).

**Comment by akx** [comment]
Anyway, the changes all seem sane -- especially since if you look at `diff -w` (or `?w=1` in GitHub parlance), it's +62/-8, so mostly whitespace changes.

Merging! Thanks, @lukas-b.

**Comment by lukas-b** [comment]
Thanks @akx ! :)

**Pull request merged**